src/com/epam/javatrainings/boundedblockingbuffer/BoundedBlockingBuffer.java

```diff
@@ -0,0 +1,46 @@
+package com.epam.javatrainings.boundedblockingbuffer;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+public class BoundedBlockingBuffer<T> {
+
+    private final Queue<T> buffer;
+    private final int capacity;
+    private boolean isFull = false;
```

Comment:
Bad Idea using a non-atomic type in a multithreading environment.

Reply:
Many thanks for useful notes !